compute/cloud-client/src/test/java/compute/reservation/ReservationIT.java

```diff
@@ -177,4 +223,50 @@ public void testCreateReservationWithRegionInstanceTemplate()
     Assert.assertTrue(reservation.getZone().contains(ZONE));
     Assert.assertEquals(RESERVATION_NAME_REGIONAL, reservation.getName());
   }
+
+  @Test
+  public void testCreateSharedReservation()
+      throws ExecutionException, InterruptedException, TimeoutException {
+    // Mock the ReservationsClient
+    ReservationsClient mockReservationsClient = mock(ReservationsClient.class);
+
+    ShareSettings shareSettings = ShareSettings.newBuilder()
+        .setShareType(String.valueOf(ShareSettings.ShareType.SPECIFIC_PROJECTS))
+        .putProjectMap("CONSUMER_PROJECT_ID_1", ShareSettingsProjectConfig.newBuilder().build())
+        .putProjectMap("CONSUMER_PROJECT_ID_2", ShareSettingsProjectConfig.newBuilder().build())
+        .build();
+
+    Reservation reservation =
+        Reservation.newBuilder()
+            .setName(RESERVATION_NAME_SHARED)
+            .setZone(ZONE)
+            .setSpecificReservationRequired(true)
+            .setShareSettings(shareSettings)
+            .setSpecificReservation(
+                AllocationSpecificSKUReservation.newBuilder()
+                    .setCount(NUMBER_OF_VMS)
+                    .setSourceInstanceTemplate(INSTANCE_TEMPLATE_SHARED_RESERV_URI)
+                    .build())
+            .build();
+
+    OperationFuture mockFuture = mock(OperationFuture.class);
+    when(mockReservationsClient.insertAsync(PROJECT_ID, ZONE, reservation))
+        .thenReturn(mockFuture);
+
+    // Mock the Operation
+    Operation mockOperation = mock(Operation.class);
+    when(mockFuture.get(3, TimeUnit.MINUTES)).thenReturn(mockOperation);
+    when(mockOperation.hasError()).thenReturn(false);
+    when(mockOperation.getStatus()).thenReturn(Operation.Status.DONE);
```

Comment:
Add explenation why we need to mock + what would have to be changes if someone wants to test this code without mocking.